2019-03-Autumn/1_linked_lists/README.md

```diff
@@ -102,4 +135,196 @@ TODO :bug:
+        Node* tmp = new Node();
+        tmp->data = sum % 10;
+        if (result == nullptr) {
+            result = tmp;
+        } else {
+            end->next = tmp;
+        }
+        end = tmp;
+    }
+    // edge case: sum requires additional decimal place
+    if (carry > 0) {
+        Node* tmp = new Node();
+        tmp->data = carry;
+        end->next = tmp;
+    }
+    return result;
+}
+```
+
+**NOTE:** This solution is iterative, and equivalent solutions may be 
+written recursively. See the detailed solution in CTCI for a recursive 
+implementation.
+
+#### Solution for Ordered List (Bonus Problem Statement)
+
+Because the order of the lists has fundamentally changed, we must take 
+a different approach. 
+
+The simplest solution will recursively traverse the two lists until the 
+last element is found, and then perform the addition at each recursive 
+level. This presents complications when we consider that the two lists 
+may have different lengths, and so our approach must compensate for this. 
+
+To compensate for this problem, we can ensure the lists are of equal 
+length by padding the shorter list with leading zeroes as needed. 
+Helper functions (or function placeholders) for this should be written.
+
+Additional complications arise when we consider how to ensure the 
+`carry` and `result` values remain consistent between recursive calls. 
+The simplest way to ensure this is to create a wrapper class/struct for 
+these specific data items, particularly because we are building a list 
+to store our results.
+
+In our implementation, our helper methods and structs look like this:
+
+```c++
+struct Node {
+    int data;
+    Node* next = nullptr;
+};
+
+struct Result {
+    Node* head = nullptr;
+    int carry = 0;
+};
+
+// Helper function
+int length(Node* n) {
+    int count = 0;
+    while (n != nullptr) {
+        count++;
+        n = n->next;
+    }
+    return count;
+}
+
+// Helper function
+Node* padListWithZeroes(Node* n, int zeroes) {
+    Node* head = n;
+    while (zeroes > 0) {
+        Node* tmp = new Node();
+        tmp->data = 0;
+        tmp->next = head;
+        head = tmp;
+        zeroes--;
+    }
+    return head;
+}
+```
+
+The core implementation is in these two methods:
+
+```c++
+// Recursive call
+void addElements(Node* n1, Node* n2, Result* result) {
+    // base case: entire list has been traversed
+    if (n1 == nullptr && n2 == nullptr) return;
+    // recurse to end of list
+    addElements(n1->next, n2->next, result);
+    // perform addition
+    int sum = n1->data + n2->data + result->carry;
+    result->carry = (sum > 9 ? 1 : 0);
+    // add sum to result list
+    Node* tmp = new Node();
+    tmp->data = sum % 10;
+    if (result->head == nullptr) result->head = tmp;
+    else {
+        tmp->next = result->head;
+        result->head = tmp;
+    }
+}
+
+// Entry point for recursive solution
+Node* getListSum(Node* n1, Node* n2) {
+    // pad list if needed
```

Comment:
function name doesn't match the problem statement

Reply:
Fixing this in the driver file for greater continuity in the README